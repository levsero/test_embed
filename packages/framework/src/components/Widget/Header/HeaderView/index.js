import styled from 'styled-components'
import { FONT_SIZE } from 'src/constants/shared'

const HeaderView = styled.header`
  padding: ${8 / FONT_SIZE}rem ${14 / FONT_SIZE}rem;
  background: ${(props) => props.theme.headerColorStr} !important;
  color: ${(props) => props.theme.headerTextColorStr} !important;
`

export default HeaderView
