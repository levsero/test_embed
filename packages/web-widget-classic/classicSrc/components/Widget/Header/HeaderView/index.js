import styled from 'styled-components'
import { FONT_SIZE } from 'classicSrc/constants/shared'

const HeaderView = styled.header`
  padding: ${8 / FONT_SIZE}rem ${14 / FONT_SIZE}rem;
  background: ${(props) => props.theme.headerColorStr} !important;
  color: ${(props) => props.theme.headerTextColorStr} !important;
`

export default HeaderView
