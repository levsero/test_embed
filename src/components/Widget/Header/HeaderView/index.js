import styled from 'styled-components'

import { FONT_SIZE } from 'constants/shared'
import { isMobileBrowser } from 'utility/devices'

const HeaderView = styled.div`
  border-radius: ${isMobileBrowser() ? 'none' : `${8 / FONT_SIZE}rem ${8 / FONT_SIZE}rem 0 0`};
  padding: ${8 / FONT_SIZE}rem ${14 / FONT_SIZE}rem;
  background: ${props => props.theme.headerColorStr} !important;
  color: ${props => props.theme.headerTextColorStr} !important;
`

export default HeaderView
