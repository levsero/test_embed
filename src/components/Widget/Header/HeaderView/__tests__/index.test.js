import React from 'react'
import { render } from '@testing-library/react'
import HeaderView from '../'
import 'jest-styled-components'

describe('HeaderView', () => {
  const defaultProps = {
    children: <div>Some child component</div>
  }

  const renderComponent = (props = {}) => render(<HeaderView {...defaultProps} {...props} />)

  it('renders', () => {
    const { container } = renderComponent()

    expect(container.firstChild).toMatchSnapshot()
  })
})
