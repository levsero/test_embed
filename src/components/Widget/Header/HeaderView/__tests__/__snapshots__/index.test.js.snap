// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderView renders 1`] = `
.c0 {
  border-radius: 0.5714285714285714rem 0.5714285714285714rem 0 0;
  padding: 0.5714285714285714rem 1rem;
  background: !important;
  color: !important;
}

<div
  class="c0"
>
  <div>
    Some child component
  </div>
</div>
`;
